src/Temporalio/Bridge/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+#![allow(
+    // We choose to have narrow "unsafe" blocks instead of marking entire
+    // functions as unsafe. Even the example in clippy's docs at
+    // https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref
+    // cause a rustc warning for unnecessary inner-unsafe when marked on fn.
+    // This check only applies to "pub" functions which are all exposed via C
+    // API.
+    clippy::not_unsafe_ptr_arg_deref,
+)]
+
+pub mod client;
+pub mod runtime;
+pub mod testing;
+
+use std::collections::HashMap;
+
+#[repr(C)]
+pub struct ByteArrayRef {
+    data: *const u8,
+    size: libc::size_t,
+}
+
+impl ByteArrayRef {
+    fn to_slice(&self) -> &[u8] {
+        unsafe { std::slice::from_raw_parts(self.data, self.size) }
+    }
+
+    fn to_vec(&self) -> Vec<u8> {
+        unsafe { std::slice::from_raw_parts(self.data, self.size).to_vec() }
+    }
+
+    fn to_str(&self) -> &str {
+        // Trust caller to send UTF8
+        unsafe { std::str::from_utf8_unchecked(std::slice::from_raw_parts(self.data, self.size)) }
```

Comment:
Note on these calls I trust the lang side to send the right thing all the time. Even if I changed this to `from_utf8` and propagated the error, lang could still mess us up in a lot of ways including just bad pointers.

Reply:
Added